components/user/user.js

```diff
@@ -0,0 +1,90 @@
+const express = require('express');
+const User = require('./usermodel.js');
+const Joi = require('joi');
+
+const router = express.Router();
+
+const validateForm = (req, res, next) => {
+  const { username, email, password } = req.body;
+
+  const user = {
+    username,
+    email,
+    password,
+  };
+
+  const schema = Joi.object().keys({
+    username: Joi.string()
+      .regex(/^[a-z0-9]{5,15}/)
+      .min(5)
+      .max(15)
+      .required(),
+    email: Joi.string()
+      .email()
+      .required(),
+    password: Joi.string()
+      .min(6)
+      .max(15)
+      .required(),
+  });
+
+  Joi.validate(user, schema, (err, value) => {
+    if (err) {
+      return res.status(400).json(err.details[0]);
+    }
+
+    if (!value) return false;
+    return next();
+  });
+};
+
+function createUser(req, done) {
+  const { username, email, password } = req.body;
+  User.find({ $or: [{ username }, { email }] })
+    .then((data) => {
+      if (data.length === 0) {
+        const user = {
+          username,
+          email,
+          password,
+        };
+
+        const newUser = new User(user);
+
+        return newUser.save((err) => {
+          if (err) return done(false, err, { message: 'Error whilst saving' });
+          return done(null, newUser);
+        });
+      }
+      const err = {
+        message: 'User exists',
+      };
+      return done(null, false, err);
+    })
+    .catch((err) => {
+      done(err);
+    });
+}
+
+/* GET home page. */
+router.post(
+  '/create',
```

Comment:
For info normally you'd POST at `/api/users` to create a new user and GET/PUT/DELETE at `/api/users/123` to read/update/delete user 123. See https://blog.mwaysolutions.com/2014/06/05/10-best-practices-for-better-restful-api/

I'm not asking for this to be changed though. We don't have time.